mmdet/core/anchor/anchor_generator.py

```diff
@@ -102,7 +255,281 @@ def valid_flags(self, featmap_size, valid_size, device='cuda'):
         valid_y[:valid_h] = 1
         valid_xx, valid_yy = meshgrid(valid_x, valid_y)
         valid = valid_xx & valid_yy
-        valid = valid[:,
-                      None].expand(valid.size(0),
-                                   self.num_base_anchors).contiguous().view(-1)
+        valid = valid[:, None].expand(valid.size(0),
+                                      num_base_anchors).contiguous().view(-1)
         return valid
+
+    def __repr__(self):
+        indent_str = '    '
+        repr_str = self.__class__.__name__ + '(\n'
+        repr_str += f'{indent_str}strides={self.strides},\n'
+        repr_str += f'{indent_str}ratios={self.ratios},\n'
+        repr_str += f'{indent_str}scales={self.scales},\n'
+        repr_str += f'{indent_str}base_sizes={self.base_sizes},\n'
+        repr_str += f'{indent_str}scale_major={self.scale_major},\n'
+        repr_str += f'{indent_str}octave_base_scale='
+        repr_str += f'{self.octave_base_scale},\n'
+        repr_str += f'{indent_str}scales_per_octave='
+        repr_str += f'{self.scales_per_octave},\n'
+        repr_str += f'{indent_str}num_levels={self.num_levels}\n'
+        repr_str += f'{indent_str}centers={self.centers},\n'
+        repr_str += f'{indent_str}center_offset={self.center_offset})'
+        return repr_str
+
+
+@ANCHOR_GENERATORS.register_module()
+class SSDAnchorGenerator(AnchorGenerator):
```

Comment:
Is it only applicable to SSD models?

Reply:
hz - it is from upstream

Reply:
Don't see `SSDAnchorGenerator` in upstream. What am I missing?

Reply:
It was in tag v2.0.0 https://github.com/open-mmlab/mmdetection/blob/v2.0.0/mmdet/core/anchor/anchor_generator.py